Automattic/jetpack
# Add support for group_id

Partially fixes 2592-gh-Automattic/p2

**This is my first contact with the JPS app, feedback is more than welcomed 🙂**

#### Changes proposed in this Pull Request:
With D58442-code we introduced API support to filter results based on a given group id. This PR attempts to add support for groups on the UI side. While they are existing search filters they are dynamic and based on the returned results while for group searching we will need to have filters that are results agnostic, I named them `static filters` but I'm bad at naming so I'm open to better suggestions.

Based on the supplied `staticFilters` JPS options "Search in" filters should be visible, at the moment we only support group searching which will send an additional `group_id` field to the API request if "All P2" is selected that would allow us to search across multiple P2s that are part of the given group.

![Markup on 2021-05-19 at 12:22:03 PM](https://user-images.githubusercontent.com/2129455/118788834-d7364880-b89c-11eb-960a-ccb2e6718446.png)

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Sandbox a "new" p2 a8c site
- Add your user ID to the `P2_EARLY_ADOPTERS` constant (find the requisite file via opengrok)
- JPS should have "Search in" filter that allows you to search the current P2 or the whole workspace
  - "All P2" should search in all sites across a8c workspace
  - Selecting the P2 name should only search that particular blog
